language_service/src/format.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::{compilation::Compilation, protocol::TextEdit};
+
+use qsc::formatter::calculate_format_edits;
+use qsc::line_column::{Encoding, Position, Range};
+
+pub(crate) fn get_format_changes(
+    compilation: &Compilation,
+    source_name: &str,
+    _position: Position,
+    encoding: Encoding,
+) -> Vec<TextEdit> {
+    let contents = compilation
+        .user_unit()
+        .sources
+        .find_by_name(source_name)
+        .expect("can't find source by name")
+        .contents
+        .clone();
+
+    calculate_format_edits(&contents)
+        .iter()
+        .map(|edit| TextEdit {
+            new_text: edit.new_text.clone(),
```

Comment:
I think you can avoid the `.clone()` here if you use `into_iter()` instead of `iter()` to move the values out of the source vector.

(Though maybe then the reference to `&edit.span` may be a problem... I wonder if Rust would allow that if you flipped the order of `new_text` and `range` so that you use `&edit.span` before you consume `edit.new_text`.)